### PR TITLE
Add utility method for clicking list items in tests #1313

### LINKED ISSUE #1313: Add utility method for clicking list items in tests
We want to make test code more high-level, so less of the implementation leaks in.

1. We should add a generic way to generate element id strings, so test code isn't filled with ad hoc constants. This should be used to generate the strings for the actual elements as well.
2. After that, we can add more general ways to interacting with UI elements without having to build the id strings manually. A good one to start with is `clickItemAt(index)` which clicks an item in a particular panel.

### PR DESCRIPTION
**Pull request by m133225**
Fixes #1313

### CODE REVIEW

**Comment by m133225** [comment]
Ready for review.

The main changes are:
- Id of ALL UI components are now gotten from the IdGenerator's static methods.
- Several clicking and getting utility methods are added to UITests, such as clickPanel, getFilterTextFieldAtPanel, rightClickIssue.
However,
- I have also implemented the methods related to the label picker's text field, but I stopped there since I thought others methods (including the label picker text field's) might be redundant. Would like some review before completely removing it.

File: src/main/java/ui/IdGenerator.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+    public static String getBoardNameSaveButtonId() {
+        return "boardsavebutton";
+    }
+
+    public static String getApiBoxId() {
+        return "apiBox";
+    }
+
+    public static String getPanelCellIdForTest(int panelIndex, int issueId) {
```
Comment (dariusf) [inline src/main/java/ui/IdGenerator.java]: We may not need the `ForTest` suffix here. These may be used in other contexts as well.

**Comment by dariusf** [comment]
Nice! Are we able to get a reference to the label picker field in tests? If so, we don't need to add any related to it. Which are the duplicated methods?

**Comment by m133225** [comment]
@darius Ready for review.
- Some minor changes to the method names `...ForReference` instead of `...ForTest`
- Slight changes to the UI components id e.g. label picker's text field is now `labelPickerTextField` instead of its fxml's `queryField`
- As discussed, the label picker's ids are still dependent on fxml's id unless specfically set

**Comment by dariusf** [comment]
Just curious, what does `ForReference` mean?

**Comment by m133225** [comment]
For reference purposes if others need to find it. :fearful: Or is there a better way to name them? ><

**Pull request merged**